stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -0,0 +1,272 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .api_client import APIClient
+import json
+
+
+class InvalidMetadataException(Exception):
+    pass
+
+
+class Connector(BaseJsonSyncConnector):
+    NOZOMI_MAX_PAGE_SIZE = 1000
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+        self.expired_token_count = 0
+
+    async def create_results_connection(self, query, offset, length, metadata=None):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :param metadata: dict
+        :return: return_obj, dict
+        """
+        data = []
+        try:
+            result_count, total_records, page_number, page_size = self.handle_metadata_and_page_size(offset, length,
+                                                                                                     metadata)
+            # Generating jwt token
+            jwt_token, return_obj = await self.__get_token()
+            if return_obj:
+                return return_obj
+
+            if (result_count == 0 and page_number == 1) or \
+                    (page_number and result_count < self.api_client.result_limit):
+
+                while result_count < total_records:
```

Comment:
request_count should always start at 0 before entering this loop. The reason is that it's a local count of how many events have been asked for from this specific "create_results_connection" request.

```suggestion
request_count = 0
while result_count < total_records:
```

Reply:
setting result_count is 0 in handle_metadata_and_page_size function.

Reply:
In this case, you are still using the remaining data from previous request, which isn't necessary.

Reply:
The remaining data count is used for slicing the data.

example 1:
The data source record count is 10000.
Input is coming as offset 0, length 1500.
Setting page_size to 1000
The first iteration fetches 1000 records.
The second iteration fetches 1000 records.
Using the remaining data count, we will slice the second iteration record to 500.
so total records return 1500.